lib/zclient.c

```diff
@@ -53,7 +53,8 @@ socklen_t zclient_addr_len;
 int zclient_debug = 0;
 
 /* Allocate zclient structure. */
-struct zclient *zclient_new(struct thread_master *master)
+struct zclient *zclient_new_notify(struct thread_master *master,
+				   bool receive_notify)
```

Comment:
I think this is a bit ugly. In the future we might want to send other parameters in the ZEBRA_HELLO message. For example, a parameter in which the client daemon requests zebra to send explicit route withdraws instead of implicit withdraws. So I'd prefer to pack all initialization parameters in a structure and pass that to `zclient_new()`. That would future-proof our API against new changes. Alternatively we could simply set these parameters by modifying the zclient directly, like we do with the callbacks.